Build/validate-qir.ps1

```diff
@@ -89,35 +104,71 @@ $qirProjects = @(
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics EvaluatingFunctions EvaluatingFunctions.csproj); Args = @() },
+    # memory leak #@{ Path = (Join-Path $PSScriptRoot .. samples numerics ResourceCounting ResourceCounting.csproj); Args = @() },
 
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation h2 command-line H2SimulationSampleCmdLine.csproj); Args = @() },
-    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation hubbard HubbardSimulationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising adiabatic AdiabaticIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising generators IsingGeneratorsSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising phase-estimation IsingPhaseEstimationSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising simple SimpleIsingSample.csproj); Args = @() },
     # not compatible #@{ Path = (Join-Path $PSScriptRoot .. samples simulation ising trotter-evolution IsingTrotterSample.csproj); Args = @() },
-    # needs argument(s) #@{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @() }
+    @{ Path = (Join-Path $PSScriptRoot .. samples simulation qaoa QAOA.csproj); Args = @("--numTrials", "20") }
 )
 
+Write-Host "##[info]Beginning Microsoft.Quantum.Qir.CommandLineTool installation."
 if (-not (Get-Command qir-cli -ErrorAction SilentlyContinue)) {
-    Write-Host "##[error]The qir-cli command is missing; you can install it by running `dotnet tool install Microsoft.Quantum.Qir.CommandLineTool -g`.";
-    $script:allOk = $False
+    $version = ($Env:NUGET_VERSION -split "-")[0] + "-alpha"
+    Write-Host "##[info]Going to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+    dotnet tool install Microsoft.Quantum.Qir.CommandLineTool --version $version -g
+    if  ($LastExitCode -ne 0) {
+        Write-Host "##[error]Failed to install Microsoft.Quantum.Qir.CommandLineTool.$version."
+        $script:allOk = $False
+    } else {
+        Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool.$version has been installed."
+    }
 } else {
+    Write-Host "##[info]The Microsoft.Quantum.Qir.CommandLineTool is already installed."
+}
+
+if ($allOk) {
+    if ($IsWindows) {
+        $old_PATH = $env:PATH
+        if (!(Get-Command clang -ErrorAction SilentlyContinue)) {
+            if ((choco find --idonly -l llvm) -notcontains "llvm") {
+                Write-Host "##[info]Installing LLVM with Chocolatey."
+                choco install llvm --version=11.1.0
```

Comment:
Similarly here, this makes a global and possibly disruptive change to the caller's environment.

Reply:
This seems like a larger design decision for how our build and build scripts work. Let's make that a separate conversation to have when all the people that are involved with builds are back. The way it is set up now is aligned with Andres' suggestions and matches what is also done on other repos. There is certainly value in pursuing a different setup, but I think then we need to schedule time to make a pass over all our repos and refactor some build scripts.